gotthard/cli.py

```diff
@@ -307,13 +305,23 @@ def get_spilo_stacks(cf, spilo_cluster):
                     'UPDATE_ROLLBACK_IN_PROGRESS', 'UPDATE_ROLLBACK_FAILED',
                     'UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS', 'UPDATE_ROLLBACK_COMPLETE']
 
-    for stack in cf.list_stacks(StackStatusFilter=stack_status)['StackSummaries']:
-        stack_name = stack['StackName']
-        if not spilo_cluster or stack_name.split('-')[-1] == spilo_cluster:
-            desc = cf.describe_stacks(StackName=stack_name)['Stacks'][0]
-            for k, v in {i['Key']: i['Value'] for i in desc['Tags']}.items():
-                if k == 'SpiloCluster' and (not spilo_cluster or v == spilo_cluster):
-                    yield stack_name
+    next_token = None
+    while True:
```

Comment:
Isn't there a more 'sparing' (ie. more direct, without listing so many stacks) to find the stack name?  That would automatically remove the problem being solved here.